src/Place/ReadModel/Duplicate/LookupDuplicatePlaceWithSapi3.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Place\ReadModel\Duplicate;
+
+use CultuurNet\UDB3\Model\Place\Place;
+use CultuurNet\UDB3\ReadModel\DocumentDoesNotExist;
+use CultuurNet\UDB3\ReadModel\DocumentRepository;
+use CultuurNet\UDB3\Search\Sapi3SearchService;
+
+class LookupDuplicatePlaceWithSapi3 implements LookupDuplicatePlace
+{
+    private Sapi3SearchService $sapi3SearchService;
+    private UniqueAddressIdentifierFactory $addressIdentifierFactory;
+    private DocumentRepository $placeRepository;
+    private string $currentUserId;
+
+    public function __construct(
+        Sapi3SearchService $sapi3SearchService,
+        UniqueAddressIdentifierFactory $addressIdentifierFactory,
+        DocumentRepository $placeRepository,
+        string $currentUserId
+    ) {
+        $this->sapi3SearchService = $sapi3SearchService;
+        $this->addressIdentifierFactory = $addressIdentifierFactory;
+        $this->placeRepository = $placeRepository;
+        $this->currentUserId = $currentUserId;
+    }
+
+    /*
+    When there is only one place propose that place
+    When there are multiple places propose the one with duplicatedBy
+    When there are multiple but no canonical returns the search query to get all places
+    */
+    public function getDuplicatePlaceUri(Place $place): ?string
+    {
+        $query = 'unique_address_identifier:' .
+            $this->addressIdentifierFactory->create(
+                $place->getTitle()->getTranslation($place->getMainLanguage())->toString(),
+                $place->getAddress()->getTranslation($place->getMainLanguage()),
+                $this->currentUserId
+            );
+
+        $results = $this->sapi3SearchService->search(
```

Comment:
If you set a limit here of `1` the total count will either be `0` or `1`, so you will never reach line 58. This needs some tweaking. It does not result in an error in the unit tests because there the search service is mocked and the limit parameter is discarded.